app/model/application.js

```diff
@@ -1,241 +1,445 @@
+  });
+
+  return Promise.filter(githubReleases, release => {
+    return semver.valid(release.tag_name, true);
+  })
+  .map(release => {
+    return application.releaseUpdateOrCreate({
+      github: {
+        id:      release.id,
+      },
+    }, {
+      github: {
+        id:      release.id,
+        author:  release.author.login,
+        date:    release.published_at,
+        tag:     release.tag_name,
+      },
+      changelog: release.body.match(/.+/g),
+    });
+  })
+  .catch(error => {
+    return Promise.reject(error);
+  })
+  .then(applications => {
+    return Application.findOne({_id: application._id});
+  });
+}
+
+// Application methods
+// Helper function for application status
+// Returns a boolean always
+ApplicationSchema.methods.isStatus = function(query) {
+  return this.status === query;
+}
+
+// Helper function for application problems
+// Returns a boolean always
+ApplicationSchema.methods.hasProblem = function(query) {
+  return _.indexOf(this.problems, query) >= 0;
+}
+
+// Sets an application problem based on status or toggles if no status
+// Returns nothing
+ApplicationSchema.methods.setProblem = function(query, status = !this.hasProblem(query)) {
+  if (status && !this.hasProblem(query)) {
+    this.problems.push(query);
+  } else if (!status && this.hasProblem(query)) {
+    this.problems = _.pull(this.problems, query);
+  }
+}
+
+// Pushes Github issue label to repository
+// Returns Promise with no data on success
+ApplicationSchema.methods.labelPush = function() {
+  const application = this;
+  const fullName = application.github.fullName;
+  const label = application.github.label;
+  app.log.silly(`application labelPush called for ${fullName}`);
+
+  return gh.request(`POST /repos/${fullName}/labels`, {
+    token: application.github.APItoken,
+    body: {
+      name: label,
+      color: '3A416F',
+    },
+  })
+  .catch(error => {
+    if (error.status === 422) { // Already created labels don't create errors
+      return Promise.resolve();
     }
-    return application;
+    return Promise.reject(error);
   });
 }
 
-ApplicationSchema.statics.parseAppHubFile = function(application) {
+// Fetches appHub file in repository
+// Returns promise of unsaved application object
+ApplicationSchema.methods.apphubFetch = function() {
+  let application = this;
   const fullName = application.github.fullName;
+  app.log.silly(`application apphubFetch called for ${fullName}`);
+
   return gh.request(`GET /repos/${fullName}/contents/.apphub`, {
     token: application.github.APItoken,
   })
-  .then(appHubFile => {
-    // Parse the .apphub file
-    const desktopFileBuf = new Buffer(appHubFile.content, 'base64');
-    const appHubFileJSON = JSON.parse(desktopFileBuf.toString())
-    application.priceUSD = appHubFileJSON.priceUSD
-    application.label = appHubFileJSON.issueLabel;
+  .catch(error => {
+    if (error.status === 404) {
+      application.setProblem('apphubExist', true);
```

Comment:
hmm, i wonder if the problem enum/title should be positive or negative? Like the problem is actually that the apphub file is *missing*... maybe the enum should reflect that?